parfive/results.py

```diff
@@ -22,14 +22,16 @@ class Results(UserList):
     """
     The results of a download from `parfive.Downloader.download`.
 
-    This object contains the filenames of successful downloads as well
-    as a list of any errors encountered in the `~parfive.Results.errors`
+    This object contains the filenames of successful downloads as well,
+    a list of all urls requested in the `~parfive.Results.urls` property
+    and a list of any errors encountered in the `~parfive.Results.errors`
     property.
     """
 
-    def __init__(self, *args, errors=None):
+    def __init__(self, *args, errors=None, urls=None):
         super().__init__(*args)
         self._errors = errors or list()
+        self._urls = urls or list()
```

Comment:
Do we want some kind of assertion that urls is the same length as the list? Or maybe something even more restrictive where we have to pair them up?